http-verbs-common/src/main/scala/uk/gov/hmrc/play/http/logging/Mdc.scala

```diff
@@ -28,11 +28,19 @@ object Mdc {
 
   def withMdc[A](block: => Future[A], mdcData: Map[String, String])(implicit ec: ExecutionContext): Future[A] =
     block.map { a =>
-      mdcData.foreach {
-        case (k, v) => MDC.put(k, v)
-      }
+      putMdc(mdcData)
       a
-    }(ec)
+    }.recover {
+      case t =>
+        putMdc(mdcData)
+        throw t
```

Comment:
Can we use `recoverWith` with Future.failed instead of re-throwing?

Reply:
Why do you prefer that?

Reply:
I think returning objects instead of throwing exceptions is a more functional approach. Just a personal preference.

Reply:
Given this is already merged, how much do you mind if we don't change the code? :)

Reply:
Ha! Don't worry about it now :) ... I am not too fussed.